### PR TITLE
Initial DDEX exchange support

### PR DESCRIPTION
**Pull request by bargst**
* public function: ticker() get_markets() get_balances()
* private function: get_orders() place_order() cancel_order()

Main missing function: get_all_trades()

### CODE REVIEW

File: pyexchange/ddex.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+
+        return order_id
+
+    def cancel_order(self, order_id: str) -> bool:
+        assert(isinstance(order_id, str))
+
+        self.logger.info(f"Cancelling order #{order_id}...")
+
+        result = self._http_delete_signed(f"/v2/orders/{order_id}", "")
+        print(result)
```
Comment (reverendus) [inline pyexchange/ddex.py]: A leftover here :P

File: pyexchange/ddex.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
+# This file is part of Maker Keeper Framework.
+#
+# Copyright (C) 2017-2018 reverendus
```
Comment (reverendus) [inline pyexchange/ddex.py]: You could put your own copyright as well :P

File: pyexchange/ddex.py [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    def __init__(self, web3: Web3, api_server: str, timeout: float):
+        assert(isinstance(web3, Web3) or (web3 is None))
+        assert(isinstance(api_server, str))
+        assert(isinstance(timeout, float))
+
+        self.web3 = web3
+        self.api_server = api_server
+        self.timeout = timeout
+        self.last_nonce = 0
+        self.last_nonce_lock = threading.Lock()
```
Comment (reverendus) [inline pyexchange/ddex.py]: These two nonce-related fields don't seem to be used, are they..?

File: pyexchange/ddex.py [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+        success = result['status']
+
+        if success == 0:
+            self.logger.info(f"Cancelled order #{order_id}")
+        else:
+            self.logger.info(f"Failed to cancel order #{order_id}")
+
+        return success
+
+    def _result(self, result, our_nonce: Optional[int] = None) -> Optional[dict]:
```
Comment (reverendus) [inline pyexchange/ddex.py]: `our_nonce` parameter is unused

**Comment by bargst** [comment]
I updated following your review

**Pull request merged**